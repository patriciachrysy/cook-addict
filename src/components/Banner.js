import React from 'react';
import PropTypes from 'prop-types';
import { RiTimerLine } from 'react-icons/ri';
import { BsPeopleFill, BsArrowRightCircle } from 'react-icons/bs';
import '../styles/Banner.css';

const Banner = (props) => {
  const { featuresRecipe } = props;
  return (
    <div className="banner">
      <div className="banner-image">
        <img src={featuresRecipe.thumbnail_url} alt={featuresRecipe.name} />
      </div>
      <div className="banner-infos">
        <div className="dish-name">
          <span>{featuresRecipe.name}</span>
        </div>
        <div className="dish-infos">
          <div>
            <span>
              {' '}
              <RiTimerLine />
              {' '}
              {featuresRecipe.prep_time_minutes}
            </span>
            <br />
            <span>
              {' '}
              <BsPeopleFill />
              {' '}
              {featuresRecipe.num_servings}
              People
            </span>
          </div>
          <button type="button">
            <BsArrowRightCircle />
          </button>
        </div>
      </div>
    </div>
  );
};

Banner.propTypes = {
  featuresRecipe: PropTypes.shape({
    thumbnail_url: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    prep_time_minutes: PropTypes.number.isRequired,
    num_servings: PropTypes.number.isRequired,
  }).isRequired,
};

export default Banner;
