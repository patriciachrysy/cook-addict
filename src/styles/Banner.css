@import url('https://fonts.googleapis.com/css2?family=Kanit:wght@500&family=Pacifico&family=Rubik:wght@300&display=swap');

.banner {
    width: 100%;
    height: 35vh;
    position: relative;
}

.banner-image {
    width: 100%;
    height: 100%;
    position: relative;
}

.banner-image img {
    width: 100%;
    height: 100%;
}

.banner-infos {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-image: linear-gradient(to right, rgba(25, 25, 25, 0.2), rgba(25, 25, 25, 0.5) , rgba(25, 25, 25, 0.8), rgba(25, 25, 25, 0.9)  );
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    color: white;
}

.dish-name {
    font-family: 'Kanit', sans-serif;
    font-size: 40px;
    align-self: end;
    padding: 5px;
}

.dish-infos {
    font-family: 'Rubik', sans-serif;
    padding: 10px;
    display: flex;
    justify-content: space-between;
    align-items: center;
}

.dish-infos div span{
    display: flex;
    align-items: center;
    font-size: 15px;
    font-weight: bold;
}

.dish-infos span svg{
    font-size: 20px;
    margin: 0 5px;
}

.dish-infos button {
    background: none;
    border: none;
    color: white;
    font-size: 25px;
}