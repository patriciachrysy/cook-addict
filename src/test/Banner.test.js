import React from 'react';
import renderer from 'react-test-renderer';

import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import Banner from '../components/Banner';

import store from '../Redux/ConfigureStore';


const data = {
    id: 1234,
    slug: 'test-recipe',
    thumbnail_url: 'https://test.com/test-recipe.png',
    name: 'test-recipe',
    prep_time_minutes: 5,
    num_servings: 18,
  }

it('renders correctly', () => {
  const tree = renderer.create(<Provider store={store}><Router><Banner featuresRecipe={data}/></Router></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render Banner component', () => {
  render(<Provider store={store}><Router><Banner featuresRecipe={data} /></Router></Provider>);
  screen.debug();
});
