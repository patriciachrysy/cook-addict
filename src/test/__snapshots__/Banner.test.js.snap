// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="banner"
>
  <div
    className="banner-image"
  >
    <img
      alt="test-recipe"
      src="https://test.com/test-recipe.png"
    />
  </div>
  <div
    className="banner-infos"
  >
    <div
      className="dish-name"
    >
      <span>
        test-recipe
      </span>
    </div>
    <div
      className="dish-infos"
    >
      <div>
        <span>
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M17.618 5.968l1.453-1.453 1.414 1.414-1.453 1.453a9 9 0 1 1-1.414-1.414zM12 20a7 7 0 1 0 0-14 7 7 0 0 0 0 14zM11 8h2v6h-2V8zM8 1h8v2H8V1z"
              />
            </g>
          </svg>
           
          5
        </span>
        <br />
        <span>
           
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 14s-1 0-1-1 1-4 5-4 5 3 5 4-1 1-1 1H7zm4-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
            />
            <path
              d="M5.216 14A2.238 2.238 0 0 1 5 13c0-1.355.68-2.75 1.936-3.72A6.325 6.325 0 0 0 5 9c-4 0-5 3-5 4s1 1 1 1h4.216z"
              fillRule="evenodd"
            />
            <path
              d="M4.5 8a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5z"
            />
          </svg>
           
          18
          People
        </span>
      </div>
      <button
        onClick={[Function]}
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            fillRule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
